sphinx/source/docs/user_guide/data.rst

```diff
@@ -0,0 +1,285 @@
+variable ``source``.
+
+Javascript
+''''''''''
+
+To create a |CustomJSFilter| with custom functionality written in JavaScript,
+pass in the JavaScript code as a string to the parameter ``code``:
+
+.. code-block:: python
+
+    custom_filter = CustomJSFilter(code='''
+    var indices = [];
+
+    // iterate through rows of data source and see if each satisfies some constraint
+    for (var i = 0; i <= source.get_length(); i++){
+        if (source.data['some_column'][i] == 'some_value'){
+            indices.push(true);
+        } else {
+            indices.push(false);
+        }
+    }
+    return indices;
+    ''')
+
+Coffeescript
+''''''''''''
+
+You can also write code for the ``CustomJSFilter`` in `CoffeeScript`_, and
+use the ``from_coffeescript`` class method, which accepts the ``code`` parameter:
+
+.. code-block:: python
+
+    custom_filter_coffee = CustomJSFilter.from_coffeescript(code='''
+    z = source.data['z']
+    indices = (i for i in [0...source.get_length()] when z[i] == 'b')
+    return indices
+    ''')
+
+Python
+''''''
+
+A CustomJSFilter can also be created from a Python function. To use this functionality
+you need the Flexx library (install with ``conda install -c bokeh flexx`` or
+``pip install flexx``). It is also important to note that not all Python code is
+supported, only the subset of Python that can be translated to Javascript using PyScript.
+
+.. code-block:: python
+
+    def custom_python_filter():
+        z = source.data['z']
+        indices = [True if z[i] == 'c' else False for i in range(len(z)) ]
+        return indices
+
+    custom_filter_pyscript = CustomJSFilter.from_py_func(custom_python_filter)
+
+For more information about the subset of Python that is supported,
+see the `PyScript documentation`_.
+
+Full example
+''''''''''''
+
+.. bokeh-plot::
+    :source-position: above
+
+    from bokeh.plotting import figure, output_file, show
+    from bokeh.layouts import row
+    from bokeh.models import ColumnDataSource, CDSView, CustomJSFilter
+
+    output_file("customjs_filter.html")
+
+    data = {'x': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+            'y': [10, 5, 2, 6, 1, 4, 8, 2, 7, 3],
+            'z': ['a', 'a', 'a', 'b', 'b', 'b', 'c', 'c', 'c', 'c']}
+
+    custom_filter = CustomJSFilter(code='''
+    var indices = [];
+    for (var i = 0; i <= source.data['z'].length; i++){
+        if (source.data['z'][i] == 'a') {
+            indices.push(true);
+        } else {
+            indices.push(false);
+        }
+
+    }
+    return indices;
+    ''')
+
+    custom_filter_coffee = CustomJSFilter.from_coffeescript(code='''
+    z = source.data['z']
+    indices = (i for i in [0...source.get_length()] when z[i] == 'b')
+    return indices
+    ''')
+
+    def custom_python_filter():
+        z = source.data['z']
+        indices = [True if z[i] == 'c' else False for i in range(len(z)) ]
+        return indices
+
+    #custom_filter_pyscript = CustomJSFilter.from_py_func(custom_python_filter)
```

Comment:
This doesn't seem to work with Sphinx? I get the following error when I uncomment this line:

```
PlotScript error:
Traceback (most recent call last):
  File "/anaconda/envs/bokeh/lib/python3.5/site-packages/flexx/pyscript/functions.py", line 48, in py2js_
    lines, linenr = inspect.getsourcelines(ob)
  File "/anaconda/envs/bokeh/lib/python3.5/inspect.py", line 930, in getsourcelines
    lines, lnum = findsource(object)
  File "/anaconda/envs/bokeh/lib/python3.5/inspect.py", line 753, in findsource
    raise OSError('source code not available')
OSError: source code not available

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/claire/Documents/bokeh/bokeh/application/handlers/code_runner.py", line 81, in run
    exec(self._code, module.__dict__)
  File "/Users/claire/Documents/bokeh/sphinx/_build/doctrees/bokeh_plot/bokeh-plot-docs-user_guide-data-inline-3.js", line 35, in <module>
  File "/Users/claire/Documents/bokeh/bokeh/models/filters.py", line 103, in from_py_func
    code = pyscript.py2js(func, 'filter') + 'return filter(%s);\n' % ', '.join(default_names)
  File "/anaconda/envs/bokeh/lib/python3.5/site-packages/flexx/pyscript/functions.py", line 83, in py2js
    return py2js_(ob)
  File "/anaconda/envs/bokeh/lib/python3.5/site-packages/flexx/pyscript/functions.py", line 51, in py2js_
    (ob, err))
ValueError: Could not get source code for object <function custom_python_filter at 0x11a34f950>: source code not available
```